{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyOdA1p6yzz6PFzdZZY/FLio"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"dYFPY2rKou_-","executionInfo":{"status":"ok","timestamp":1762406990925,"user_tz":-330,"elapsed":34484,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"daa6fbc0-5d31-420a-fa9e-efad52175f9d"},"outputs":[{"output_type":"stream","name":"stdout","text":["\r0% [Working]\r            \rGet:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n","\r0% [Waiting for headers] [Waiting for headers] [1 InRelease 0 B/3,632 B 0%] [Wa\r                                                                               \rGet:2 https://cli.github.com/packages stable InRelease [3,917 B]\n","Get:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n","Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n","Get:5 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n","Get:6 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n","Get:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [2,103 kB]\n","Get:8 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n","Hit:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n","Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n","Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n","Get:12 https://cli.github.com/packages stable/main amd64 Packages [344 B]\n","Get:13 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n","Get:14 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,421 kB]\n","Get:15 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,820 kB]\n","Get:16 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,520 kB]\n","Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,855 kB]\n","Get:18 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,288 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,594 kB]\n","Fetched 25.0 MB in 3s (9,015 kB/s)\n","Reading package lists... Done\n","W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following additional packages will be installed:\n","  blends-common ca-certificates-java default-jre default-jre-headless\n","  fonts-dejavu-core fonts-dejavu-extra java-common libapache-pom-java\n","  libargs4j-java libatk-wrapper-java libatk-wrapper-java-jni\n","  libcommons-compress-java libcommons-io-java libcommons-jexl2-java\n","  libcommons-lang3-java libcommons-logging-java libcommons-math3-java\n","  libcommons-parent-java libfindbin-libs-perl libhts3 libhtscodecs2\n","  libhtsjdk-java libjbzip2-java libjson-simple-java libkdf5-2 libncbi-vdb2\n","  libncbi-wvdb2 libngs-java libngs-sdk-dev libngs-sdk2 libpcsclite1\n","  libsis-base-java libsis-base-jni libsis-jhdf5-java libsis-jhdf5-jni\n","  libsnappy-java libsnappy-jni libxtst6 libxxf86dga1 med-config menu\n","  openjdk-11-jre openjdk-11-jre-headless x11-utils\n","Suggested packages:\n","  python3-numpy python3-matplotlib texlive-latex-recommended blends-doc\n","  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc\n","  libavalon-framework-java libcommons-logging-java-doc\n","  libexcalibur-logkit-java liblog4j1.2-java picard-tools libjson-simple-doc\n","  pcscd menu-l10n gksu | kde-runtime | ktsuss libnss-mdns fonts-ipafont-gothic\n","  fonts-ipafont-mincho fonts-wqy-microhei | fonts-wqy-zenhei fonts-indic\n","  cwltool mesa-utils\n","The following NEW packages will be installed:\n","  bcftools blends-common bwa ca-certificates-java default-jre\n","  default-jre-headless fastqc fonts-dejavu-core fonts-dejavu-extra java-common\n","  libapache-pom-java libargs4j-java libatk-wrapper-java\n","  libatk-wrapper-java-jni libcommons-compress-java libcommons-io-java\n","  libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n","  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl libhts3\n","  libhtscodecs2 libhtsjdk-java libjbzip2-java libjson-simple-java libkdf5-2\n","  libncbi-vdb2 libncbi-wvdb2 libngs-java libngs-sdk-dev libngs-sdk2\n","  libpcsclite1 libsis-base-java libsis-base-jni libsis-jhdf5-java\n","  libsis-jhdf5-jni libsnappy-java libsnappy-jni libxtst6 libxxf86dga1\n","  med-config menu openjdk-11-jre openjdk-11-jre-headless samtools sra-toolkit\n","  x11-utils\n","0 upgraded, 49 newly installed, 0 to remove and 47 not upgraded.\n","Need to get 64.4 MB of archives.\n","After this operation, 229 MB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n","Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bcftools amd64 1.13-1 [697 kB]\n","Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n","Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n","Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bwa amd64 0.7.17-6 [195 kB]\n","Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 java-common all 0.72build2 [6,782 B]\n","Get:8 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libpcsclite1 amd64 1.9.5-3ubuntu1 [19.8 kB]\n","Get:9 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre-headless amd64 11.0.28+6-1ubuntu1~22.04.1 [42.6 MB]\n","Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n","Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 ca-certificates-java all 20190909ubuntu1.2 [12.1 kB]\n","Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n","Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.28+6-1ubuntu1~22.04.1 [214 kB]\n","Get:14 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n","Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n","Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n","Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n","Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n","Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n","Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n","Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n","Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n","Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n","Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n","Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n","Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n","Get:28 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n","Get:29 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n","Get:30 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n","Get:31 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n","Get:32 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n","Get:33 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n","Get:34 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n","Get:35 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n","Get:36 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n","Get:37 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n","Get:38 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n","Get:39 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n","Get:40 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n","Get:41 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n","Get:42 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n","Get:43 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n","Get:44 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n","Get:45 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n","Get:46 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n","Get:47 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n","Get:48 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n","Get:49 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n","Fetched 64.4 MB in 1s (44.1 MB/s)\n","Extracting templates from packages: 100%\n","Preconfiguring packages ...\n","Selecting previously unselected package libhtscodecs2:amd64.\n","(Reading database ... 125081 files and directories currently installed.)\n","Preparing to unpack .../00-libhtscodecs2_1.1.1-3_amd64.deb ...\n","Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n","Selecting previously unselected package libhts3:amd64.\n","Preparing to unpack .../01-libhts3_1.13+ds-2build1_amd64.deb ...\n","Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n","Selecting previously unselected package bcftools.\n","Preparing to unpack .../02-bcftools_1.13-1_amd64.deb ...\n","Unpacking bcftools (1.13-1) ...\n","Selecting previously unselected package menu.\n","Preparing to unpack .../03-menu_2.1.47ubuntu4_amd64.deb ...\n","Unpacking menu (2.1.47ubuntu4) ...\n","Selecting previously unselected package blends-common.\n","Preparing to unpack .../04-blends-common_0.7.4ubuntu1_all.deb ...\n","Unpacking blends-common (0.7.4ubuntu1) ...\n","Selecting previously unselected package bwa.\n","Preparing to unpack .../05-bwa_0.7.17-6_amd64.deb ...\n","Unpacking bwa (0.7.17-6) ...\n","Selecting previously unselected package java-common.\n","Preparing to unpack .../06-java-common_0.72build2_all.deb ...\n","Unpacking java-common (0.72build2) ...\n","Selecting previously unselected package libpcsclite1:amd64.\n","Preparing to unpack .../07-libpcsclite1_1.9.5-3ubuntu1_amd64.deb ...\n","Unpacking libpcsclite1:amd64 (1.9.5-3ubuntu1) ...\n","Selecting previously unselected package openjdk-11-jre-headless:amd64.\n","Preparing to unpack .../08-openjdk-11-jre-headless_11.0.28+6-1ubuntu1~22.04.1_amd64.deb ...\n","Unpacking openjdk-11-jre-headless:amd64 (11.0.28+6-1ubuntu1~22.04.1) ...\n","Selecting previously unselected package default-jre-headless.\n","Preparing to unpack .../09-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n","Unpacking default-jre-headless (2:1.11-72build2) ...\n","Selecting previously unselected package ca-certificates-java.\n","Preparing to unpack .../10-ca-certificates-java_20190909ubuntu1.2_all.deb ...\n","Unpacking ca-certificates-java (20190909ubuntu1.2) ...\n","Selecting previously unselected package libxtst6:amd64.\n","Preparing to unpack .../11-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n","Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n","Selecting previously unselected package openjdk-11-jre:amd64.\n","Preparing to unpack .../12-openjdk-11-jre_11.0.28+6-1ubuntu1~22.04.1_amd64.deb ...\n","Unpacking openjdk-11-jre:amd64 (11.0.28+6-1ubuntu1~22.04.1) ...\n","Selecting previously unselected package default-jre.\n","Preparing to unpack .../13-default-jre_2%3a1.11-72build2_amd64.deb ...\n","Unpacking default-jre (2:1.11-72build2) ...\n","Selecting previously unselected package libcommons-math3-java.\n","Preparing to unpack .../14-libcommons-math3-java_3.6.1-3_all.deb ...\n","Unpacking libcommons-math3-java (3.6.1-3) ...\n","Selecting previously unselected package libjbzip2-java.\n","Preparing to unpack .../15-libjbzip2-java_0.9.1-7_all.deb ...\n","Unpacking libjbzip2-java (0.9.1-7) ...\n","Selecting previously unselected package libapache-pom-java.\n","Preparing to unpack .../16-libapache-pom-java_18-1_all.deb ...\n","Unpacking libapache-pom-java (18-1) ...\n","Selecting previously unselected package libcommons-parent-java.\n","Preparing to unpack .../17-libcommons-parent-java_43-1_all.deb ...\n","Unpacking libcommons-parent-java (43-1) ...\n","Selecting previously unselected package libcommons-compress-java.\n","Preparing to unpack .../18-libcommons-compress-java_1.21-1_all.deb ...\n","Unpacking libcommons-compress-java (1.21-1) ...\n","Selecting previously unselected package libcommons-logging-java.\n","Preparing to unpack .../19-libcommons-logging-java_1.2-2_all.deb ...\n","Unpacking libcommons-logging-java (1.2-2) ...\n","Selecting previously unselected package libcommons-jexl2-java.\n","Preparing to unpack .../20-libcommons-jexl2-java_2.1.1-5_all.deb ...\n","Unpacking libcommons-jexl2-java (2.1.1-5) ...\n","Selecting previously unselected package libjson-simple-java.\n","Preparing to unpack .../21-libjson-simple-java_2.3.0-1_all.deb ...\n","Unpacking libjson-simple-java (2.3.0-1) ...\n","Selecting previously unselected package libngs-sdk2:amd64.\n","Preparing to unpack .../22-libngs-sdk2_2.11.2-3_amd64.deb ...\n","Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n","Selecting previously unselected package libngs-sdk-dev:amd64.\n","Preparing to unpack .../23-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n","Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n","Selecting previously unselected package libngs-java:amd64.\n","Preparing to unpack .../24-libngs-java_2.11.2-3_amd64.deb ...\n","Unpacking libngs-java:amd64 (2.11.2-3) ...\n","Selecting previously unselected package libsnappy-jni.\n","Preparing to unpack .../25-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n","Unpacking libsnappy-jni (1.1.8.3-1) ...\n","Selecting previously unselected package libsnappy-java.\n","Preparing to unpack .../26-libsnappy-java_1.1.8.3-1_all.deb ...\n","Unpacking libsnappy-java (1.1.8.3-1) ...\n","Selecting previously unselected package libhtsjdk-java.\n","Preparing to unpack .../27-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n","Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n","Selecting previously unselected package libfindbin-libs-perl.\n","Preparing to unpack .../28-libfindbin-libs-perl_2.190.02-1_all.deb ...\n","Unpacking libfindbin-libs-perl (2.190.02-1) ...\n","Selecting previously unselected package libargs4j-java.\n","Preparing to unpack .../29-libargs4j-java_2.33-1.1_all.deb ...\n","Unpacking libargs4j-java (2.33-1.1) ...\n","Selecting previously unselected package libcommons-io-java.\n","Preparing to unpack .../30-libcommons-io-java_2.11.0-2_all.deb ...\n","Unpacking libcommons-io-java (2.11.0-2) ...\n","Selecting previously unselected package libcommons-lang3-java.\n","Preparing to unpack .../31-libcommons-lang3-java_3.11-1_all.deb ...\n","Unpacking libcommons-lang3-java (3.11-1) ...\n","Selecting previously unselected package libsis-base-jni.\n","Preparing to unpack .../32-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n","Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n","Selecting previously unselected package libsis-base-java.\n","Preparing to unpack .../33-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n","Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n","Selecting previously unselected package libsis-jhdf5-jni.\n","Preparing to unpack .../34-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n","Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n","Selecting previously unselected package libsis-jhdf5-java.\n","Preparing to unpack .../35-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n","Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n","Selecting previously unselected package fastqc.\n","Preparing to unpack .../36-fastqc_0.11.9+dfsg-5_all.deb ...\n","Unpacking fastqc (0.11.9+dfsg-5) ...\n","Selecting previously unselected package fonts-dejavu-core.\n","Preparing to unpack .../37-fonts-dejavu-core_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-core (2.37-2build1) ...\n","Selecting previously unselected package fonts-dejavu-extra.\n","Preparing to unpack .../38-fonts-dejavu-extra_2.37-2build1_all.deb ...\n","Unpacking fonts-dejavu-extra (2.37-2build1) ...\n","Selecting previously unselected package libxxf86dga1:amd64.\n","Preparing to unpack .../39-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n","Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Selecting previously unselected package x11-utils.\n","Preparing to unpack .../40-x11-utils_7.7+5build2_amd64.deb ...\n","Unpacking x11-utils (7.7+5build2) ...\n","Selecting previously unselected package libatk-wrapper-java.\n","Preparing to unpack .../41-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n","Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n","Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n","Preparing to unpack .../42-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n","Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Selecting previously unselected package libkdf5-2.\n","Preparing to unpack .../43-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n","Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n","Selecting previously unselected package libncbi-vdb2.\n","Preparing to unpack .../44-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n","Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n","Selecting previously unselected package libncbi-wvdb2.\n","Preparing to unpack .../45-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n","Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n","Selecting previously unselected package samtools.\n","Preparing to unpack .../46-samtools_1.13-4_amd64.deb ...\n","Unpacking samtools (1.13-4) ...\n","Selecting previously unselected package sra-toolkit.\n","Preparing to unpack .../47-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n","Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n","Selecting previously unselected package med-config.\n","Preparing to unpack .../48-med-config_3.7.1_all.deb ...\n","Unpacking med-config (3.7.1) ...\n","Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n","Setting up libhts3:amd64 (1.13+ds-2build1) ...\n","Setting up java-common (0.72build2) ...\n","Setting up bwa (0.7.17-6) ...\n","Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n","Setting up bcftools (1.13-1) ...\n","Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n","Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n","Setting up libfindbin-libs-perl (2.190.02-1) ...\n","Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n","Setting up libjson-simple-java (2.3.0-1) ...\n","Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n","Setting up samtools (1.13-4) ...\n","Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n","Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n","Setting up libapache-pom-java (18-1) ...\n","Setting up fonts-dejavu-core (2.37-2build1) ...\n","Setting up libjbzip2-java (0.9.1-7) ...\n","Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n","Setting up libpcsclite1:amd64 (1.9.5-3ubuntu1) ...\n","Setting up fonts-dejavu-extra (2.37-2build1) ...\n","Setting up x11-utils (7.7+5build2) ...\n","Setting up libcommons-math3-java (3.6.1-3) ...\n","Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n","Setting up libatk-wrapper-java (0.38.0-5build1) ...\n","Setting up libngs-java:amd64 (2.11.2-3) ...\n","Setting up menu (2.1.47ubuntu4) ...\n","Setting up libsnappy-jni (1.1.8.3-1) ...\n","Setting up libargs4j-java (2.33-1.1) ...\n","Setting up blends-common (0.7.4ubuntu1) ...\n","Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n","Setting up med-config (3.7.1) ...\n","Adding group `med' (GID 107) ...\n","Done.\n","Setting up libcommons-parent-java (43-1) ...\n","Setting up libcommons-logging-java (1.2-2) ...\n","Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n","Setting up libcommons-jexl2-java (2.1.1-5) ...\n","Setting up libcommons-lang3-java (3.11-1) ...\n","Setting up libsnappy-java (1.1.8.3-1) ...\n","Setting up libcommons-compress-java (1.21-1) ...\n","Setting up libcommons-io-java (2.11.0-2) ...\n","Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n","Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n","Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n","Setting up default-jre-headless (2:1.11-72build2) ...\n","Setting up openjdk-11-jre-headless:amd64 (11.0.28+6-1ubuntu1~22.04.1) ...\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/jjs to provide /usr/bin/jjs (jjs) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/rmid to provide /usr/bin/rmid (rmid) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/pack200 to provide /usr/bin/pack200 (pack200) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/unpack200 to provide /usr/bin/unpack200 (unpack200) in auto mode\n","update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode\n","Setting up openjdk-11-jre:amd64 (11.0.28+6-1ubuntu1~22.04.1) ...\n","Setting up default-jre (2:1.11-72build2) ...\n","Setting up ca-certificates-java (20190909ubuntu1.2) ...\n","head: cannot open '/etc/ssl/certs/java/cacerts' for reading: No such file or directory\n","Adding debian:Starfield_Services_Root_Certificate_Authority_-_G2.pem\n","Adding debian:Certum_Trusted_Root_CA.pem\n","Adding debian:DigiCert_TLS_ECC_P384_Root_G5.pem\n","Adding debian:Amazon_Root_CA_1.pem\n","Adding debian:Security_Communication_ECC_RootCA1.pem\n","Adding debian:CA_Disig_Root_R2.pem\n","Adding debian:certSIGN_Root_CA_G2.pem\n","Adding debian:XRamp_Global_CA_Root.pem\n","Adding debian:GlobalSign_ECC_Root_CA_-_R5.pem\n","Adding debian:HARICA_TLS_ECC_Root_CA_2021.pem\n","Adding debian:GlobalSign_Root_R46.pem\n","Adding debian:COMODO_Certification_Authority.pem\n","Adding debian:Hongkong_Post_Root_CA_3.pem\n","Adding debian:GLOBALTRUST_2020.pem\n","Adding debian:ISRG_Root_X2.pem\n","Adding debian:USERTrust_RSA_Certification_Authority.pem\n","Adding debian:D-TRUST_EV_Root_CA_1_2020.pem\n","Adding debian:GlobalSign_Root_CA_-_R3.pem\n","Adding debian:ANF_Secure_Server_Root_CA.pem\n","Adding debian:DigiCert_Assured_ID_Root_G2.pem\n","Adding debian:vTrus_ECC_Root_CA.pem\n","Adding debian:SSL.com_Root_Certification_Authority_ECC.pem\n","Adding debian:Entrust_Root_Certification_Authority.pem\n","Adding debian:OISTE_WISeKey_Global_Root_GB_CA.pem\n","Adding debian:Security_Communication_RootCA2.pem\n","Adding debian:Entrust_Root_Certification_Authority_-_G4.pem\n","Adding debian:Starfield_Class_2_CA.pem\n","Adding debian:TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.pem\n","Adding debian:Buypass_Class_2_Root_CA.pem\n","Adding debian:vTrus_Root_CA.pem\n","Adding debian:AC_RAIZ_FNMT-RCM_SERVIDORES_SEGUROS.pem\n","Adding debian:QuoVadis_Root_CA_3.pem\n","Adding debian:SSL.com_EV_Root_Certification_Authority_ECC.pem\n","Adding debian:Hellenic_Academic_and_Research_Institutions_RootCA_2015.pem\n","Adding debian:SwissSign_Gold_CA_-_G2.pem\n","Adding debian:UCA_Global_G2_Root.pem\n","Adding debian:NAVER_Global_Root_Certification_Authority.pem\n","Adding debian:TunTrust_Root_CA.pem\n","Adding debian:USERTrust_ECC_Certification_Authority.pem\n","Adding debian:D-TRUST_BR_Root_CA_1_2020.pem\n","Adding debian:NetLock_Arany_=Class_Gold=_Főtanúsítvány.pem\n","Adding debian:Certainly_Root_E1.pem\n","Adding debian:Certigna_Root_CA.pem\n","Adding debian:GDCA_TrustAUTH_R5_ROOT.pem\n","Adding debian:Amazon_Root_CA_4.pem\n","Adding debian:HARICA_TLS_RSA_Root_CA_2021.pem\n","Adding debian:D-TRUST_Root_Class_3_CA_2_2009.pem\n","Adding debian:DigiCert_TLS_RSA4096_Root_G5.pem\n","Adding debian:Microsoft_RSA_Root_Certificate_Authority_2017.pem\n","Adding debian:GlobalSign_Root_E46.pem\n","Adding debian:Security_Communication_Root_CA.pem\n","Adding debian:TeliaSonera_Root_CA_v1.pem\n","Adding debian:emSign_ECC_Root_CA_-_G3.pem\n","Adding debian:Microsec_e-Szigno_Root_CA_2009.pem\n","Adding debian:Security_Communication_RootCA3.pem\n","Adding debian:DigiCert_Global_Root_G2.pem\n","Adding debian:Entrust_Root_Certification_Authority_-_G2.pem\n","Adding debian:AffirmTrust_Networking.pem\n","Adding debian:GTS_Root_R4.pem\n","Adding debian:Baltimore_CyberTrust_Root.pem\n","Adding debian:Amazon_Root_CA_3.pem\n","Adding debian:SwissSign_Silver_CA_-_G2.pem\n","Adding debian:Trustwave_Global_ECC_P256_Certification_Authority.pem\n","Adding debian:Comodo_AAA_Services_root.pem\n","Adding debian:HiPKI_Root_CA_-_G1.pem\n","Adding debian:CFCA_EV_ROOT.pem\n","Adding debian:T-TeleSec_GlobalRoot_Class_2.pem\n","Adding debian:Go_Daddy_Class_2_CA.pem\n","Adding debian:DigiCert_High_Assurance_EV_Root_CA.pem\n","Adding debian:GTS_Root_R3.pem\n","Adding debian:SecureSign_RootCA11.pem\n","Adding debian:TWCA_Global_Root_CA.pem\n","Adding debian:certSIGN_ROOT_CA.pem\n","Adding debian:Certum_Trusted_Network_CA_2.pem\n","Adding debian:DigiCert_Global_Root_G3.pem\n","Adding debian:GTS_Root_R1.pem\n","Adding debian:AffirmTrust_Commercial.pem\n","Adding debian:COMODO_RSA_Certification_Authority.pem\n","Adding debian:SZAFIR_ROOT_CA2.pem\n","Adding debian:Certum_EC-384_CA.pem\n","Adding debian:Izenpe.com.pem\n","Adding debian:QuoVadis_Root_CA_3_G3.pem\n","Adding debian:e-Szigno_Root_CA_2017.pem\n","Adding debian:Microsoft_ECC_Root_Certificate_Authority_2017.pem\n","Adding debian:Atos_TrustedRoot_2011.pem\n","Adding debian:GlobalSign_Root_CA.pem\n","Adding debian:emSign_ECC_Root_CA_-_C3.pem\n","Adding debian:DigiCert_Assured_ID_Root_CA.pem\n","Adding debian:Telia_Root_CA_v2.pem\n","Adding debian:DigiCert_Trusted_Root_G4.pem\n","Adding debian:Starfield_Root_Certificate_Authority_-_G2.pem\n","Adding debian:Certainly_Root_R1.pem\n","Adding debian:emSign_Root_CA_-_G1.pem\n","Adding debian:DigiCert_Global_Root_CA.pem\n","Adding debian:COMODO_ECC_Certification_Authority.pem\n","Adding debian:SSL.com_Root_Certification_Authority_RSA.pem\n","Adding debian:Entrust_Root_Certification_Authority_-_EC1.pem\n","Adding debian:T-TeleSec_GlobalRoot_Class_3.pem\n","Adding debian:AffirmTrust_Premium_ECC.pem\n","Adding debian:Amazon_Root_CA_2.pem\n","Adding debian:ISRG_Root_X1.pem\n","Adding debian:AC_RAIZ_FNMT-RCM.pem\n","Adding debian:GlobalSign_ECC_Root_CA_-_R4.pem\n","Adding debian:GTS_Root_R2.pem\n","Adding debian:IdenTrust_Commercial_Root_CA_1.pem\n","Adding debian:UCA_Extended_Validation_Root.pem\n","Adding debian:emSign_Root_CA_-_C1.pem\n","Adding debian:TWCA_Root_Certification_Authority.pem\n","Adding debian:Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.pem\n","Adding debian:IdenTrust_Public_Sector_Root_CA_1.pem\n","Adding debian:Actalis_Authentication_Root_CA.pem\n","Adding debian:DigiCert_Assured_ID_Root_G3.pem\n","Adding debian:AffirmTrust_Premium.pem\n","Adding debian:Entrust.net_Premium_2048_Secure_Server_CA.pem\n","Adding debian:Secure_Global_CA.pem\n","Adding debian:D-TRUST_Root_Class_3_CA_2_EV_2009.pem\n","Adding debian:Go_Daddy_Root_Certificate_Authority_-_G2.pem\n","Adding debian:ePKI_Root_Certification_Authority.pem\n","Adding debian:GlobalSign_Root_CA_-_R6.pem\n","Adding debian:Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.pem\n","Adding debian:ACCVRAIZ1.pem\n","Adding debian:SecureTrust_CA.pem\n","Adding debian:QuoVadis_Root_CA_1_G3.pem\n","Adding debian:SSL.com_EV_Root_Certification_Authority_RSA_R2.pem\n","Adding debian:OISTE_WISeKey_Global_Root_GC_CA.pem\n","Adding debian:Trustwave_Global_ECC_P384_Certification_Authority.pem\n","Adding debian:Buypass_Class_3_Root_CA.pem\n","Adding debian:Trustwave_Global_Certification_Authority.pem\n","Adding debian:Certum_Trusted_Network_CA.pem\n","Adding debian:QuoVadis_Root_CA_2_G3.pem\n","Adding debian:Certigna.pem\n","Adding debian:QuoVadis_Root_CA_2.pem\n","Adding debian:SSL.com_TLS_RSA_Root_CA_2022.pem\n","Adding debian:BJCA_Global_Root_CA1.pem\n","Adding debian:CommScope_Public_Trust_RSA_Root-01.pem\n","Adding debian:Atos_TrustedRoot_Root_CA_RSA_TLS_2021.pem\n","Adding debian:Sectigo_Public_Server_Authentication_Root_R46.pem\n","Adding debian:Sectigo_Public_Server_Authentication_Root_E46.pem\n","Adding debian:CommScope_Public_Trust_ECC_Root-02.pem\n","Adding debian:SSL.com_TLS_ECC_Root_CA_2022.pem\n","Adding debian:TrustAsia_Global_Root_CA_G4.pem\n","Adding debian:CommScope_Public_Trust_ECC_Root-01.pem\n","Adding debian:CommScope_Public_Trust_RSA_Root-02.pem\n","Adding debian:TrustAsia_Global_Root_CA_G3.pem\n","Adding debian:BJCA_Global_Root_CA2.pem\n","Adding debian:Atos_TrustedRoot_Root_CA_ECC_TLS_2021.pem\n","done.\n","Setting up fastqc (0.11.9+dfsg-5) ...\n","Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n","\n","Processing triggers for man-db (2.10.2-1) ...\n","Processing triggers for ca-certificates (20240203~22.04.1) ...\n","Updating certificates in /etc/ssl/certs...\n","0 added, 0 removed; done.\n","Running hooks in /etc/ca-certificates/update.d...\n","\n","done.\n","done.\n","Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n","Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n","Processing triggers for hicolor-icon-theme (0.17-2) ...\n","Processing triggers for menu (2.1.47ubuntu4) ...\n"]}],"source":["#  Step 1: Install Required Tools\n","!apt-get update\n","!apt-get install -y sra-toolkit bwa samtools bcftools fastqc"]},{"cell_type":"code","source":["# Step 2: Download FASTQ Data for SRR9200893\n","!prefetch SRR9200893\n","!fastq-dump --split-files SRR9200893"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"w_Lnvv3-pJ_f","executionInfo":{"status":"ok","timestamp":1762407059623,"user_tz":-330,"elapsed":41858,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"62e2c7b6-b58a-4a45-f4f8-3d2cedd76b70"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","2025-11-06T05:30:24 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n","2025-11-06T05:30:24 prefetch.2.11.3: 1) Downloading 'SRR9200893'...\n","2025-11-06T05:30:24 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n","2025-11-06T05:30:24 prefetch.2.11.3:  Downloading via HTTPS...\n","2025-11-06T05:30:33 prefetch.2.11.3:  HTTPS download succeed\n","2025-11-06T05:30:35 prefetch.2.11.3:  'SRR9200893' is valid\n","2025-11-06T05:30:35 prefetch.2.11.3: 1) 'SRR9200893' was downloaded successfully\n","Read 2000000 spots for SRR9200893\n","Written 2000000 spots for SRR9200893\n"]}]},{"cell_type":"code","source":["# Step 3: Run FastQC\n","!fastqc SRR9200893_1.fastq -o /content/\n","!fastqc SRR9200893_2.fastq -o /content/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"cY4l_nTZpoxE","executionInfo":{"status":"ok","timestamp":1762407186576,"user_tz":-330,"elapsed":41000,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"1e33dbad-9ebc-470a-df19-139f0788232c"},"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["Started analysis of SRR9200893_1.fastq\n","Approx 5% complete for SRR9200893_1.fastq\n","Approx 10% complete for SRR9200893_1.fastq\n","Approx 15% complete for SRR9200893_1.fastq\n","Approx 20% complete for SRR9200893_1.fastq\n","Approx 25% complete for SRR9200893_1.fastq\n","Approx 30% complete for SRR9200893_1.fastq\n","Approx 35% complete for SRR9200893_1.fastq\n","Approx 40% complete for SRR9200893_1.fastq\n","Approx 45% complete for SRR9200893_1.fastq\n","Approx 50% complete for SRR9200893_1.fastq\n","Approx 55% complete for SRR9200893_1.fastq\n","Approx 60% complete for SRR9200893_1.fastq\n","Approx 65% complete for SRR9200893_1.fastq\n","Approx 70% complete for SRR9200893_1.fastq\n","Approx 75% complete for SRR9200893_1.fastq\n","Approx 80% complete for SRR9200893_1.fastq\n","Approx 85% complete for SRR9200893_1.fastq\n","Approx 90% complete for SRR9200893_1.fastq\n","Approx 95% complete for SRR9200893_1.fastq\n","Approx 100% complete for SRR9200893_1.fastq\n","Analysis complete for SRR9200893_1.fastq\n","Started analysis of SRR9200893_2.fastq\n","Approx 5% complete for SRR9200893_2.fastq\n","Approx 10% complete for SRR9200893_2.fastq\n","Approx 15% complete for SRR9200893_2.fastq\n","Approx 20% complete for SRR9200893_2.fastq\n","Approx 25% complete for SRR9200893_2.fastq\n","Approx 30% complete for SRR9200893_2.fastq\n","Approx 35% complete for SRR9200893_2.fastq\n","Approx 40% complete for SRR9200893_2.fastq\n","Approx 45% complete for SRR9200893_2.fastq\n","Approx 50% complete for SRR9200893_2.fastq\n","Approx 55% complete for SRR9200893_2.fastq\n","Approx 60% complete for SRR9200893_2.fastq\n","Approx 65% complete for SRR9200893_2.fastq\n","Approx 70% complete for SRR9200893_2.fastq\n","Approx 75% complete for SRR9200893_2.fastq\n","Approx 80% complete for SRR9200893_2.fastq\n","Approx 85% complete for SRR9200893_2.fastq\n","Approx 90% complete for SRR9200893_2.fastq\n","Approx 95% complete for SRR9200893_2.fastq\n","Approx 100% complete for SRR9200893_2.fastq\n","Analysis complete for SRR9200893_2.fastq\n"]}]},{"cell_type":"code","source":["# Step 4: Download and Index Human Reference Genome (GRCh38)\n","!wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_44/GRCh38.primary_assembly.genome.fa.gz\n","!gunzip GRCh38.primary_assembly.genome.fa.gz\n","!bwa index GRCh38.primary_assembly.genome.fa"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"0fvyHWf2p6J3","executionInfo":{"status":"ok","timestamp":1762413200386,"user_tz":-330,"elapsed":5990215,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"bf93ed60-d354-4f72-a631-8f6d07590bd5"},"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":["--2025-11-06 05:33:36--  ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_44/GRCh38.primary_assembly.genome.fa.gz\n","           => ‘GRCh38.primary_assembly.genome.fa.gz’\n","Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.193.165\n","Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.193.165|:21... connected.\n","Error in server response. Closing.\n","Retrying.\n","\n","--2025-11-06 05:33:37--  ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_44/GRCh38.primary_assembly.genome.fa.gz\n","  (try: 2) => ‘GRCh38.primary_assembly.genome.fa.gz’\n","Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.193.165|:21... connected.\n","Logging in as anonymous ... Logged in!\n","==> SYST ... done.    ==> PWD ... done.\n","==> TYPE I ... done.  ==> CWD (1) /pub/databases/gencode/Gencode_human/release_44 ... done.\n","==> SIZE GRCh38.primary_assembly.genome.fa.gz ... 844691642\n","==> PASV ... done.    ==> RETR GRCh38.primary_assembly.genome.fa.gz ... done.\n","Length: 844691642 (806M) (unauthoritative)\n","\n","GRCh38.primary_asse 100%[===================>] 805.56M  50.1MB/s    in 17s     \n","\n","2025-11-06 05:33:55 (47.0 MB/s) - ‘GRCh38.primary_assembly.genome.fa.gz’ saved [844691642]\n","\n","[bwa_index] Pack FASTA... 25.71 sec\n","[bwa_index] Construct BWT for the packed sequence...\n","[BWTIncCreate] textLength=6199501436, availableWord=448219384\n","[BWTIncConstructFromPacked] 10 iterations done. 99999996 characters processed.\n","[BWTIncConstructFromPacked] 20 iterations done. 199999996 characters processed.\n","[BWTIncConstructFromPacked] 30 iterations done. 299999996 characters processed.\n","[BWTIncConstructFromPacked] 40 iterations done. 399999996 characters processed.\n","[BWTIncConstructFromPacked] 50 iterations done. 499999996 characters processed.\n","[BWTIncConstructFromPacked] 60 iterations done. 599999996 characters processed.\n","[BWTIncConstructFromPacked] 70 iterations done. 699999996 characters processed.\n","[BWTIncConstructFromPacked] 80 iterations done. 799999996 characters processed.\n","[BWTIncConstructFromPacked] 90 iterations done. 899999996 characters processed.\n","[BWTIncConstructFromPacked] 100 iterations done. 999999996 characters processed.\n","[BWTIncConstructFromPacked] 110 iterations done. 1099999996 characters processed.\n","[BWTIncConstructFromPacked] 120 iterations done. 1199999996 characters processed.\n","[BWTIncConstructFromPacked] 130 iterations done. 1299999996 characters processed.\n","[BWTIncConstructFromPacked] 140 iterations done. 1399999996 characters processed.\n","[BWTIncConstructFromPacked] 150 iterations done. 1499999996 characters processed.\n","[BWTIncConstructFromPacked] 160 iterations done. 1599999996 characters processed.\n","[BWTIncConstructFromPacked] 170 iterations done. 1699999996 characters processed.\n","[BWTIncConstructFromPacked] 180 iterations done. 1799999996 characters processed.\n","[BWTIncConstructFromPacked] 190 iterations done. 1899999996 characters processed.\n","[BWTIncConstructFromPacked] 200 iterations done. 1999999996 characters processed.\n","[BWTIncConstructFromPacked] 210 iterations done. 2099999996 characters processed.\n","[BWTIncConstructFromPacked] 220 iterations done. 2199999996 characters processed.\n","[BWTIncConstructFromPacked] 230 iterations done. 2299999996 characters processed.\n","[BWTIncConstructFromPacked] 240 iterations done. 2399999996 characters processed.\n","[BWTIncConstructFromPacked] 250 iterations done. 2499999996 characters processed.\n","[BWTIncConstructFromPacked] 260 iterations done. 2599999996 characters processed.\n","[BWTIncConstructFromPacked] 270 iterations done. 2699999996 characters processed.\n","[BWTIncConstructFromPacked] 280 iterations done. 2799999996 characters processed.\n","[BWTIncConstructFromPacked] 290 iterations done. 2899999996 characters processed.\n","[BWTIncConstructFromPacked] 300 iterations done. 2999999996 characters processed.\n","[BWTIncConstructFromPacked] 310 iterations done. 3099999996 characters processed.\n","[BWTIncConstructFromPacked] 320 iterations done. 3199999996 characters processed.\n","[BWTIncConstructFromPacked] 330 iterations done. 3299999996 characters processed.\n","[BWTIncConstructFromPacked] 340 iterations done. 3399999996 characters processed.\n","[BWTIncConstructFromPacked] 350 iterations done. 3499999996 characters processed.\n","[BWTIncConstructFromPacked] 360 iterations done. 3599999996 characters processed.\n","[BWTIncConstructFromPacked] 370 iterations done. 3699999996 characters processed.\n","[BWTIncConstructFromPacked] 380 iterations done. 3799999996 characters processed.\n","[BWTIncConstructFromPacked] 390 iterations done. 3899999996 characters processed.\n","[BWTIncConstructFromPacked] 400 iterations done. 3999999996 characters processed.\n","[BWTIncConstructFromPacked] 410 iterations done. 4099999996 characters processed.\n","[BWTIncConstructFromPacked] 420 iterations done. 4199999996 characters processed.\n","[BWTIncConstructFromPacked] 430 iterations done. 4299999996 characters processed.\n","[BWTIncConstructFromPacked] 440 iterations done. 4399999996 characters processed.\n","[BWTIncConstructFromPacked] 450 iterations done. 4499999996 characters processed.\n","[BWTIncConstructFromPacked] 460 iterations done. 4599999996 characters processed.\n","[BWTIncConstructFromPacked] 470 iterations done. 4699999996 characters processed.\n","[BWTIncConstructFromPacked] 480 iterations done. 4799999996 characters processed.\n","[BWTIncConstructFromPacked] 490 iterations done. 4899999996 characters processed.\n","[BWTIncConstructFromPacked] 500 iterations done. 4999999996 characters processed.\n","[BWTIncConstructFromPacked] 510 iterations done. 5099999996 characters processed.\n","[BWTIncConstructFromPacked] 520 iterations done. 5199999996 characters processed.\n","[BWTIncConstructFromPacked] 530 iterations done. 5299999996 characters processed.\n","[BWTIncConstructFromPacked] 540 iterations done. 5399999996 characters processed.\n","[BWTIncConstructFromPacked] 550 iterations done. 5499999996 characters processed.\n","[BWTIncConstructFromPacked] 560 iterations done. 5596733548 characters processed.\n","[BWTIncConstructFromPacked] 570 iterations done. 5683010028 characters processed.\n","[BWTIncConstructFromPacked] 580 iterations done. 5759688796 characters processed.\n","[BWTIncConstructFromPacked] 590 iterations done. 5827837068 characters processed.\n","[BWTIncConstructFromPacked] 600 iterations done. 5888403388 characters processed.\n","[BWTIncConstructFromPacked] 610 iterations done. 5942230876 characters processed.\n","[BWTIncConstructFromPacked] 620 iterations done. 5990068876 characters processed.\n","[BWTIncConstructFromPacked] 630 iterations done. 6032583388 characters processed.\n","[BWTIncConstructFromPacked] 640 iterations done. 6070366396 characters processed.\n","[BWTIncConstructFromPacked] 650 iterations done. 6103943980 characters processed.\n","[BWTIncConstructFromPacked] 660 iterations done. 6133783772 characters processed.\n","[BWTIncConstructFromPacked] 670 iterations done. 6160301452 characters processed.\n","[BWTIncConstructFromPacked] 680 iterations done. 6183866396 characters processed.\n","[bwt_gen] Finished constructing BWT in 688 iterations.\n","[bwa_index] 3801.57 seconds elapse.\n","[bwa_index] Update BWT... 20.42 sec\n","[bwa_index] Pack forward-only FASTA... 16.65 sec\n","[bwa_index] Construct SA from BWT and Occ... 1882.68 sec\n","[main] Version: 0.7.17-r1188\n","[main] CMD: bwa index GRCh38.primary_assembly.genome.fa\n","[main] Real time: 5931.210 sec; CPU: 5747.034 sec\n"]}]},{"cell_type":"code","source":["# Step 5: Align Reads to Genome\n","!bwa mem GRCh38.primary_assembly.genome.fa SRR9200893_1.fastq SRR9200893_2.fastq > aligned.sam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"41R68D3A73C1","executionInfo":{"status":"ok","timestamp":1762414716935,"user_tz":-330,"elapsed":1472443,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"3a9aab29-6909-4e86-98d8-009510d2d701"},"execution_count":10,"outputs":[{"output_type":"stream","name":"stdout","text":["[M::bwa_idx_load_from_disk] read 0 ALT contigs\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 30565, 12, 21)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (170, 221, 566)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1358)\n","[M::mem_pestat] mean and std.dev: (326.58, 264.39)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1754)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (126, 188, 355)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 813)\n","[M::mem_pestat] mean and std.dev: (198.51, 128.00)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1042)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (492, 805, 2287)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5877)\n","[M::mem_pestat] mean and std.dev: (1137.27, 872.22)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7672)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (270, 410, 1531)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4053)\n","[M::mem_pestat] mean and std.dev: (938.95, 1189.22)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5696)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.920 CPU sec, 35.458 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 30307, 16, 23)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (235, 418, 1121)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2893)\n","[M::mem_pestat] mean and std.dev: (617.17, 553.30)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3779)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (131, 198, 377)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 869)\n","[M::mem_pestat] mean and std.dev: (207.47, 134.49)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1115)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (237, 662, 3911)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11259)\n","[M::mem_pestat] mean and std.dev: (1542.50, 1884.39)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 14933)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (211, 388, 1290)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3448)\n","[M::mem_pestat] mean and std.dev: (469.37, 453.05)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4527)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.580 CPU sec, 34.847 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (34, 30233, 22, 22)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (116, 422, 1415)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4013)\n","[M::mem_pestat] mean and std.dev: (730.52, 890.12)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5312)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (132, 201, 385)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 891)\n","[M::mem_pestat] mean and std.dev: (212.24, 140.59)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1144)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (352, 517, 874)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1918)\n","[M::mem_pestat] mean and std.dev: (503.58, 369.49)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2440)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (178, 655, 2568)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7348)\n","[M::mem_pestat] mean and std.dev: (1428.80, 1847.60)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 9738)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.355 CPU sec, 34.515 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 30135, 9, 26)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (128, 269, 1890)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5414)\n","[M::mem_pestat] mean and std.dev: (686.80, 1016.58)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7176)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (133, 203, 387)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 895)\n","[M::mem_pestat] mean and std.dev: (212.16, 138.63)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1149)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (224, 357, 1039)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2669)\n","[M::mem_pestat] mean and std.dev: (535.23, 544.99)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3484)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.886 CPU sec, 35.099 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (30, 30323, 17, 25)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (138, 389, 1049)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2871)\n","[M::mem_pestat] mean and std.dev: (624.48, 657.62)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3782)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (132, 201, 383)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 885)\n","[M::mem_pestat] mean and std.dev: (211.25, 138.18)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1136)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (183, 248, 1097)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2925)\n","[M::mem_pestat] mean and std.dev: (429.07, 379.12)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3839)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (171, 353, 1158)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3132)\n","[M::mem_pestat] mean and std.dev: (666.38, 687.39)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4119)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 35.369 CPU sec, 35.556 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 30407, 13, 18)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (191, 231, 1255)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3383)\n","[M::mem_pestat] mean and std.dev: (567.95, 594.59)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4447)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (134, 204, 376)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 860)\n","[M::mem_pestat] mean and std.dev: (212.47, 135.06)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1102)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (311, 392, 832)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1874)\n","[M::mem_pestat] mean and std.dev: (493.54, 368.40)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2395)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (104, 213, 798)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2186)\n","[M::mem_pestat] mean and std.dev: (304.07, 276.04)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2880)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.607 CPU sec, 38.029 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (37, 30222, 17, 22)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (132, 199, 689)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1803)\n","[M::mem_pestat] mean and std.dev: (313.19, 373.81)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2360)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (135, 207, 384)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 882)\n","[M::mem_pestat] mean and std.dev: (215.47, 136.58)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1131)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (227, 363, 2290)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6416)\n","[M::mem_pestat] mean and std.dev: (1034.41, 1303.20)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8479)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (147, 341, 1271)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3519)\n","[M::mem_pestat] mean and std.dev: (561.63, 746.35)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4643)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 36.714 CPU sec, 37.160 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (31, 30276, 19, 30)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (192, 222, 460)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 996)\n","[M::mem_pestat] mean and std.dev: (295.82, 213.87)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1264)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (135, 205, 378)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 864)\n","[M::mem_pestat] mean and std.dev: (213.59, 133.17)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1107)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (403, 955, 1919)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4951)\n","[M::mem_pestat] mean and std.dev: (1267.28, 1262.08)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6467)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (124, 291, 1279)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3589)\n","[M::mem_pestat] mean and std.dev: (650.55, 814.01)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4744)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.943 CPU sec, 38.171 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (34, 30142, 23, 26)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (142, 219, 437)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1027)\n","[M::mem_pestat] mean and std.dev: (268.03, 215.50)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1322)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (134, 200, 361)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 815)\n","[M::mem_pestat] mean and std.dev: (208.09, 124.30)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1042)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (272, 495, 1095)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2741)\n","[M::mem_pestat] mean and std.dev: (579.52, 517.97)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3564)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (148, 274, 601)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1507)\n","[M::mem_pestat] mean and std.dev: (281.10, 217.66)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1960)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 42.403 CPU sec, 42.798 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 30364, 19, 31)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (109, 289, 2277)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6613)\n","[M::mem_pestat] mean and std.dev: (955.05, 1308.65)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8781)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (118, 174, 317)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 715)\n","[M::mem_pestat] mean and std.dev: (183.10, 111.85)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 914)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (457, 670, 3408)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9310)\n","[M::mem_pestat] mean and std.dev: (1444.89, 1658.24)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 12261)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (135, 251, 1596)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4518)\n","[M::mem_pestat] mean and std.dev: (691.82, 996.53)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5979)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.942 CPU sec, 34.090 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (26, 30560, 13, 20)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (130, 262, 760)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2020)\n","[M::mem_pestat] mean and std.dev: (297.73, 261.01)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2650)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (123, 183, 335)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 759)\n","[M::mem_pestat] mean and std.dev: (192.05, 120.09)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 971)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (352, 725, 930)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2086)\n","[M::mem_pestat] mean and std.dev: (554.91, 249.56)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2664)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (167, 664, 1451)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4019)\n","[M::mem_pestat] mean and std.dev: (576.11, 539.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5303)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 32.933 CPU sec, 33.130 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 30467, 12, 30)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (142, 291, 1575)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4441)\n","[M::mem_pestat] mean and std.dev: (861.39, 1005.57)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5874)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (126, 189, 349)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 795)\n","[M::mem_pestat] mean and std.dev: (197.29, 123.99)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1018)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (163, 329, 639)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1591)\n","[M::mem_pestat] mean and std.dev: (363.82, 284.79)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2067)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (199, 597, 4488)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13066)\n","[M::mem_pestat] mean and std.dev: (2254.90, 2862.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 17355)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.514 CPU sec, 33.696 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (31, 30190, 12, 28)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (115, 255, 523)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1339)\n","[M::mem_pestat] mean and std.dev: (270.00, 256.58)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1747)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (129, 193, 362)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 828)\n","[M::mem_pestat] mean and std.dev: (202.03, 127.83)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (193, 399, 805)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2029)\n","[M::mem_pestat] mean and std.dev: (515.58, 454.71)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2641)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (85, 319, 4462)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13216)\n","[M::mem_pestat] mean and std.dev: (2119.61, 2849.77)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 17593)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.935 CPU sec, 34.141 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30395, 18, 31)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (106, 223, 925)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2563)\n","[M::mem_pestat] mean and std.dev: (446.22, 529.80)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3382)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (129, 194, 363)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 831)\n","[M::mem_pestat] mean and std.dev: (203.05, 129.35)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1065)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (184, 423, 824)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2104)\n","[M::mem_pestat] mean and std.dev: (428.75, 337.81)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2744)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (118, 289, 1476)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4192)\n","[M::mem_pestat] mean and std.dev: (527.79, 663.19)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5550)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.335 CPU sec, 34.545 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30268, 17, 24)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (179, 261, 477)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1073)\n","[M::mem_pestat] mean and std.dev: (326.83, 245.03)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1371)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (132, 201, 374)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 858)\n","[M::mem_pestat] mean and std.dev: (208.81, 132.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1100)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (148, 358, 1227)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3385)\n","[M::mem_pestat] mean and std.dev: (631.00, 774.47)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4464)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (256, 695, 1910)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5218)\n","[M::mem_pestat] mean and std.dev: (1095.23, 1351.30)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6872)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 35.394 CPU sec, 35.519 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (27, 30128, 30, 34)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (151, 315, 1524)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4270)\n","[M::mem_pestat] mean and std.dev: (804.68, 1090.78)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5643)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (134, 201, 364)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 824)\n","[M::mem_pestat] mean and std.dev: (208.79, 127.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1054)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (342, 627, 1392)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3492)\n","[M::mem_pestat] mean and std.dev: (706.23, 583.73)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4542)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (105, 227, 758)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2064)\n","[M::mem_pestat] mean and std.dev: (435.78, 527.97)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2717)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 41.814 CPU sec, 42.190 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 30199, 19, 26)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (183, 318, 913)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2373)\n","[M::mem_pestat] mean and std.dev: (437.88, 444.27)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3103)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (123, 185, 340)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 774)\n","[M::mem_pestat] mean and std.dev: (194.42, 120.19)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 991)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (753, 1299, 4553)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12153)\n","[M::mem_pestat] mean and std.dev: (2194.16, 2060.02)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 15953)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (197, 365, 2578)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7340)\n","[M::mem_pestat] mean and std.dev: (1570.96, 2028.16)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 9721)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 38.323 CPU sec, 38.470 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30658, 14, 22)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (88, 312, 1647)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4765)\n","[M::mem_pestat] mean and std.dev: (486.71, 683.02)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6324)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (114, 163, 292)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 648)\n","[M::mem_pestat] mean and std.dev: (171.42, 100.53)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 826)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (234, 487, 1273)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3351)\n","[M::mem_pestat] mean and std.dev: (578.17, 515.60)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4390)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (82, 184, 523)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1405)\n","[M::mem_pestat] mean and std.dev: (230.00, 200.77)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1846)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 31.232 CPU sec, 31.375 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30610, 18, 25)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (116, 200, 326)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 746)\n","[M::mem_pestat] mean and std.dev: (198.14, 108.15)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 956)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (118, 173, 310)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 694)\n","[M::mem_pestat] mean and std.dev: (180.15, 106.48)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 886)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (249, 488, 1742)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4728)\n","[M::mem_pestat] mean and std.dev: (1022.00, 1167.96)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6221)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (152, 276, 1043)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2825)\n","[M::mem_pestat] mean and std.dev: (559.78, 727.66)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3716)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 31.787 CPU sec, 31.956 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 30466, 18, 36)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (210, 264, 788)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1944)\n","[M::mem_pestat] mean and std.dev: (458.29, 454.24)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2522)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (123, 181, 320)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 714)\n","[M::mem_pestat] mean and std.dev: (186.78, 110.46)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 911)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (77, 311, 772)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2162)\n","[M::mem_pestat] mean and std.dev: (298.00, 257.93)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2857)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (115, 392, 2217)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6421)\n","[M::mem_pestat] mean and std.dev: (949.32, 1187.32)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8523)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 32.710 CPU sec, 32.966 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30271, 22, 28)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (120, 389, 1625)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4635)\n","[M::mem_pestat] mean and std.dev: (476.38, 623.10)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6140)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (125, 186, 337)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 761)\n","[M::mem_pestat] mean and std.dev: (192.56, 116.44)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 973)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (213, 520, 754)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1836)\n","[M::mem_pestat] mean and std.dev: (425.44, 277.00)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2377)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (172, 293, 737)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1867)\n","[M::mem_pestat] mean and std.dev: (320.21, 237.34)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2432)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.226 CPU sec, 33.431 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 30466, 18, 25)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (190, 448, 1331)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3613)\n","[M::mem_pestat] mean and std.dev: (598.63, 589.23)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4754)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (130, 196, 357)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 811)\n","[M::mem_pestat] mean and std.dev: (202.27, 123.95)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1038)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (262, 679, 898)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2170)\n","[M::mem_pestat] mean and std.dev: (587.00, 320.34)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2806)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (123, 378, 1271)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3567)\n","[M::mem_pestat] mean and std.dev: (621.17, 759.78)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4715)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.710 CPU sec, 34.867 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (34, 29735, 21, 28)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (141, 303, 1264)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3510)\n","[M::mem_pestat] mean and std.dev: (572.00, 672.55)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4633)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (133, 201, 365)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 829)\n","[M::mem_pestat] mean and std.dev: (206.63, 125.52)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1061)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (277, 459, 1512)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3982)\n","[M::mem_pestat] mean and std.dev: (804.35, 836.27)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5217)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (228, 458, 1829)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5031)\n","[M::mem_pestat] mean and std.dev: (1070.43, 1236.66)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6632)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.909 CPU sec, 38.076 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (23, 30321, 17, 19)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (126, 267, 1905)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5463)\n","[M::mem_pestat] mean and std.dev: (939.23, 1195.51)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7242)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (131, 200, 383)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 887)\n","[M::mem_pestat] mean and std.dev: (210.32, 138.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1139)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (486, 571, 909)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1755)\n","[M::mem_pestat] mean and std.dev: (520.79, 231.15)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2178)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (207, 410, 1781)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4929)\n","[M::mem_pestat] mean and std.dev: (985.94, 1033.59)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6503)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.434 CPU sec, 34.649 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 30149, 14, 31)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (161, 783, 1082)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2924)\n","[M::mem_pestat] mean and std.dev: (616.24, 414.80)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3845)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (134, 205, 397)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 923)\n","[M::mem_pestat] mean and std.dev: (217.47, 145.90)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1186)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (353, 734, 2277)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6125)\n","[M::mem_pestat] mean and std.dev: (1009.31, 1030.93)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8049)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (120, 270, 1513)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4299)\n","[M::mem_pestat] mean and std.dev: (701.61, 1106.95)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5692)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.715 CPU sec, 34.977 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (31, 30061, 17, 33)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (116, 246, 1220)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3428)\n","[M::mem_pestat] mean and std.dev: (554.43, 655.31)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4532)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (136, 212, 418)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 982)\n","[M::mem_pestat] mean and std.dev: (226.64, 157.83)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1264)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (373, 994, 3337)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9265)\n","[M::mem_pestat] mean and std.dev: (1567.41, 1668.12)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 12229)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (167, 413, 1768)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4970)\n","[M::mem_pestat] mean and std.dev: (756.53, 970.12)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6571)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 35.757 CPU sec, 35.969 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (14, 30243, 24, 16)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (150, 346, 670)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1710)\n","[M::mem_pestat] mean and std.dev: (467.00, 459.16)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2304)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (139, 214, 422)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 988)\n","[M::mem_pestat] mean and std.dev: (228.62, 160.43)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1271)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (325, 614, 1949)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5197)\n","[M::mem_pestat] mean and std.dev: (904.55, 860.72)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6821)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (263, 704, 3380)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9614)\n","[M::mem_pestat] mean and std.dev: (2004.69, 2556.38)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 12731)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.896 CPU sec, 38.442 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 30253, 20, 33)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (106, 231, 472)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1204)\n","[M::mem_pestat] mean and std.dev: (210.35, 172.49)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1570)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (136, 212, 414)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 970)\n","[M::mem_pestat] mean and std.dev: (225.09, 154.91)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1248)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (257, 639, 994)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2468)\n","[M::mem_pestat] mean and std.dev: (508.53, 310.68)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3205)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (174, 307, 2388)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6816)\n","[M::mem_pestat] mean and std.dev: (1103.38, 1432.99)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 9030)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 38.001 CPU sec, 38.517 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (32, 30205, 14, 32)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (120, 281, 1094)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3042)\n","[M::mem_pestat] mean and std.dev: (460.76, 447.16)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4016)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (137, 213, 411)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 959)\n","[M::mem_pestat] mean and std.dev: (225.13, 151.41)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1233)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (423, 743, 3076)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8382)\n","[M::mem_pestat] mean and std.dev: (1626.29, 1877.01)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 11035)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (298, 747, 1999)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5401)\n","[M::mem_pestat] mean and std.dev: (1257.81, 1317.31)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7102)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.434 CPU sec, 37.646 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 30119, 22, 26)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (113, 483, 1180)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3314)\n","[M::mem_pestat] mean and std.dev: (656.55, 753.56)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4381)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (138, 214, 410)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 954)\n","[M::mem_pestat] mean and std.dev: (226.32, 151.17)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1226)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (385, 763, 2562)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6916)\n","[M::mem_pestat] mean and std.dev: (1312.23, 1113.74)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 9093)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (109, 291, 839)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2299)\n","[M::mem_pestat] mean and std.dev: (489.12, 525.08)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3029)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 38.611 CPU sec, 38.817 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 30217, 19, 24)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (216, 670, 2234)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6270)\n","[M::mem_pestat] mean and std.dev: (910.09, 870.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8288)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (140, 211, 385)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 875)\n","[M::mem_pestat] mean and std.dev: (219.58, 135.28)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1120)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (494, 1175, 4826)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13490)\n","[M::mem_pestat] mean and std.dev: (2324.26, 2446.41)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 17822)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (121, 239, 472)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1174)\n","[M::mem_pestat] mean and std.dev: (236.29, 202.64)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1525)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 42.071 CPU sec, 42.651 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (22, 30587, 13, 33)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (81, 236, 1530)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4428)\n","[M::mem_pestat] mean and std.dev: (733.10, 1044.07)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5877)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (121, 178, 333)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 757)\n","[M::mem_pestat] mean and std.dev: (188.21, 117.83)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 969)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (201, 379, 3636)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10506)\n","[M::mem_pestat] mean and std.dev: (1420.08, 1972.09)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 13941)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (159, 343, 2860)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8262)\n","[M::mem_pestat] mean and std.dev: (1532.84, 2282.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 10963)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.681 CPU sec, 33.895 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (22, 30451, 9, 20)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (183, 343, 1223)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3303)\n","[M::mem_pestat] mean and std.dev: (506.32, 599.85)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4343)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (126, 189, 356)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 816)\n","[M::mem_pestat] mean and std.dev: (199.35, 127.29)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1046)\n","[M::mem_pestat] skip orientation RF as there are not enough pairs\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (124, 158, 546)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1390)\n","[M::mem_pestat] mean and std.dev: (186.69, 128.89)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1812)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 33.978 CPU sec, 34.146 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 30443, 13, 27)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (198, 878, 3890)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 11274)\n","[M::mem_pestat] mean and std.dev: (2099.21, 2373.82)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 14966)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (130, 196, 365)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 835)\n","[M::mem_pestat] mean and std.dev: (204.14, 128.70)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1070)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (550, 673, 3093)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8179)\n","[M::mem_pestat] mean and std.dev: (1382.77, 1348.80)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 10722)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (173, 247, 1529)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4241)\n","[M::mem_pestat] mean and std.dev: (539.65, 681.96)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 5597)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 34.746 CPU sec, 34.950 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (19, 30223, 14, 27)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (186, 653, 1790)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4998)\n","[M::mem_pestat] mean and std.dev: (647.50, 634.78)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 6602)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (133, 204, 397)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 925)\n","[M::mem_pestat] mean and std.dev: (216.87, 147.18)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1189)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (202, 544, 703)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1705)\n","[M::mem_pestat] mean and std.dev: (474.54, 367.32)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2206)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (126, 236, 756)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2016)\n","[M::mem_pestat] mean and std.dev: (374.21, 456.56)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2646)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 35.453 CPU sec, 35.645 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (32, 30471, 17, 17)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (115, 242, 1048)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2914)\n","[M::mem_pestat] mean and std.dev: (595.12, 753.43)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 3847)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (134, 205, 397)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 923)\n","[M::mem_pestat] mean and std.dev: (218.72, 147.64)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1186)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (166, 469, 3117)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9019)\n","[M::mem_pestat] mean and std.dev: (1396.29, 1845.88)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 11970)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (380, 871, 2240)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5960)\n","[M::mem_pestat] mean and std.dev: (1227.19, 1036.98)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7820)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 35.760 CPU sec, 35.943 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (18, 30339, 18, 17)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (173, 437, 1332)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3650)\n","[M::mem_pestat] mean and std.dev: (626.81, 821.22)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 4809)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (136, 208, 399)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 925)\n","[M::mem_pestat] mean and std.dev: (219.33, 144.73)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1188)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (176, 681, 2308)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 6572)\n","[M::mem_pestat] mean and std.dev: (1118.81, 1396.77)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 8704)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (236, 359, 3193)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 9107)\n","[M::mem_pestat] mean and std.dev: (1466.76, 1562.78)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 12064)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 36.253 CPU sec, 36.439 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (30, 30172, 14, 24)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (129, 306, 796)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2130)\n","[M::mem_pestat] mean and std.dev: (567.59, 607.62)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2998)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (138, 211, 406)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 942)\n","[M::mem_pestat] mean and std.dev: (224.64, 148.99)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1210)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (413, 823, 2094)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5456)\n","[M::mem_pestat] mean and std.dev: (1251.23, 1303.89)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7137)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (174, 394, 2909)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8379)\n","[M::mem_pestat] mean and std.dev: (1162.78, 1724.70)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 11114)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 37.336 CPU sec, 37.566 real sec\n","[M::process] read 100000 sequences (10000000 bp)...\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (10, 27440, 49, 24)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (123, 265, 546)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1392)\n","[M::mem_pestat] mean and std.dev: (252.50, 172.54)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1815)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (120, 179, 316)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 708)\n","[M::mem_pestat] mean and std.dev: (186.39, 109.40)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 904)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (241, 1050, 2824)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7990)\n","[M::mem_pestat] mean and std.dev: (1436.65, 1574.97)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 10573)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (103, 202, 502)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1300)\n","[M::mem_pestat] mean and std.dev: (190.95, 133.46)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1699)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 42.638 CPU sec, 43.139 real sec\n","[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (26, 30779, 12, 16)\n","[M::mem_pestat] analyzing insert size distribution for orientation FF...\n","[M::mem_pestat] (25, 50, 75) percentile: (118, 273, 655)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1729)\n","[M::mem_pestat] mean and std.dev: (352.50, 414.42)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 2266)\n","[M::mem_pestat] analyzing insert size distribution for orientation FR...\n","[M::mem_pestat] (25, 50, 75) percentile: (117, 170, 305)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 681)\n","[M::mem_pestat] mean and std.dev: (177.48, 105.72)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 869)\n","[M::mem_pestat] analyzing insert size distribution for orientation RF...\n","[M::mem_pestat] (25, 50, 75) percentile: (346, 565, 2024)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5380)\n","[M::mem_pestat] mean and std.dev: (1206.00, 1516.25)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 7271)\n","[M::mem_pestat] analyzing insert size distribution for orientation RR...\n","[M::mem_pestat] (25, 50, 75) percentile: (135, 212, 432)\n","[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1026)\n","[M::mem_pestat] mean and std.dev: (226.36, 157.14)\n","[M::mem_pestat] low and high boundaries for proper pairs: (1, 1323)\n","[M::mem_pestat] skip orientation FF\n","[M::mem_pestat] skip orientation RF\n","[M::mem_pestat] skip orientation RR\n","[M::mem_process_seqs] Processed 100000 reads in 32.146 CPU sec, 32.536 real sec\n","[main] Version: 0.7.17-r1188\n","[main] CMD: bwa mem GRCh38.primary_assembly.genome.fa SRR9200893_1.fastq SRR9200893_2.fastq\n","[main] Real time: 1472.138 sec; CPU: 1441.290 sec\n"]}]},{"cell_type":"code","source":["#  Step 6: Convert SAM to BAM, Sort and Index\n","!samtools view -Sb aligned.sam > aligned.bam\n","!samtools sort aligned.bam -o sorted.bam\n","!samtools index sorted.bam"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"rnpCUjQJ764p","executionInfo":{"status":"ok","timestamp":1762414843720,"user_tz":-330,"elapsed":86766,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"44ce5ed7-1e9c-49d9-dca6-0b888d474608"},"execution_count":12,"outputs":[{"output_type":"stream","name":"stdout","text":["[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n"]}]},{"cell_type":"code","source":["#  Step 7: Variant Calling\n","!samtools mpileup -uf GRCh38.primary_assembly.genome.fa sorted.bam | bcftools call -mv -Ov -o variants.vcf"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"b8CRehI579pf","executionInfo":{"status":"ok","timestamp":1762415223603,"user_tz":-330,"elapsed":360032,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"06bed75c-8a32-45c2-be0a-9f1295ef13b5"},"execution_count":14,"outputs":[{"output_type":"stream","name":"stdout","text":["[warning] samtools mpileup option `u` is functional, but deprecated. Please switch to using bcftools mpileup in future.\n","Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n","[mpileup] 1 samples in 1 input files\n"]}]},{"cell_type":"code","source":["# Step 8: Preview First 20 Lines of VCF File\n","with open('variants.vcf') as f:\n","    for _ in range(20):\n","        print(f.readline().strip())\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"3amcD6bH8CT5","executionInfo":{"status":"ok","timestamp":1762415234834,"user_tz":-330,"elapsed":33,"user":{"displayName":"Dr. Nagendra","userId":"05632517937859890145"}},"outputId":"374a1737-2450-4e1e-b549-b07c96c133cc"},"execution_count":15,"outputs":[{"output_type":"stream","name":"stdout","text":["##fileformat=VCFv4.2\n","##FILTER=<ID=PASS,Description=\"All filters passed\">\n","##samtoolsVersion=1.13+htslib-1.13+ds\n","##samtoolsCommand=samtools mpileup -uf GRCh38.primary_assembly.genome.fa sorted.bam\n","##reference=file://GRCh38.primary_assembly.genome.fa\n","##contig=<ID=chr1,length=248956422>\n","##contig=<ID=chr2,length=242193529>\n","##contig=<ID=chr3,length=198295559>\n","##contig=<ID=chr4,length=190214555>\n","##contig=<ID=chr5,length=181538259>\n","##contig=<ID=chr6,length=170805979>\n","##contig=<ID=chr7,length=159345973>\n","##contig=<ID=chr8,length=145138636>\n","##contig=<ID=chr9,length=138394717>\n","##contig=<ID=chr10,length=133797422>\n","##contig=<ID=chr11,length=135086622>\n","##contig=<ID=chr12,length=133275309>\n","##contig=<ID=chr13,length=114364328>\n","##contig=<ID=chr14,length=107043718>\n","##contig=<ID=chr15,length=101991189>\n"]}]},{"cell_type":"markdown","metadata":{"id":"99fd92da"},"source":["Here is a detailed, line-by-line interpretation of the findings in your `variants.vcf` file, based on the first 20 lines you previewed:\n","\n","*   **`##fileformat=VCFv4.2`**: This line specifies the version of the Variant Call Format (VCF) used in this file, which is version 4.2.\n","*   **`##FILTER=<ID=PASS,Description=\"All filters passed\">`**: This line defines a FILTER that indicates whether a variant has passed all quality filters. In this case, a variant with `FILTER=PASS` is considered a high-quality call.\n","*   **`##samtoolsVersion=1.13+htslib-1.13+ds`**: This indicates the version of Samtools used to generate part of this VCF file (specifically, the mpileup step).\n","*   **`##samtoolsCommand=samtools mpileup -uf GRCh38.primary_assembly.genome.fa sorted.bam`**: This line records the exact `samtools mpileup` command used, including the reference genome and the sorted BAM file.\n","*   **`##reference=file://GRCh38.primary_assembly.genome.fa`**: This specifies the path to the reference genome file used for alignment and variant calling.\n","*   **`##contig=<ID=chr1,length=248956422>`** (and similar lines for other chromosomes): These lines define the contigs (chromosomes or sequences) present in the reference genome and their lengths. This information is important for locating variants within the genome.\n","*   Subsequent lines (not fully shown in the preview but expected in a VCF) would represent the actual variant calls. Each of these lines typically starts with chromosome and position information and includes details about the reference allele, alternate allele(s), quality score, and other relevant information about the variant and the sample.\n","\n","In summary, the initial lines of the VCF file provide metadata about the file format, the tools and commands used, the reference genome, and the chromosomes it contains. The lines that follow (not fully shown in the preview) contain the specific details of each identified genetic variation."]},{"cell_type":"markdown","metadata":{"id":"cfd4a3bf"},"source":["Here is a summary of the steps we have completed in this notebook:\n","\n","1.  **Installed Required Tools**: We installed essential bioinformatics tools including sra-toolkit, bwa, samtools, bcftools, and fastqc.\n","2.  **Downloaded FASTQ Data**: We downloaded sequencing data for the SRA accession SRR9200893 using `prefetch` and extracted the FASTQ files using `fastq-dump`.\n","3.  **Ran FastQC**: We performed quality control on the downloaded FASTQ files using `fastqc` to assess the quality of the sequencing reads.\n","4.  **Downloaded and Indexed Human Reference Genome**: We downloaded the GRCh38 human reference genome and indexed it using `bwa index` for alignment.\n","5.  **Aligned Reads to Genome**: We aligned the FASTQ reads to the indexed reference genome using `bwa mem`, generating a SAM file.\n","6.  **Converted SAM to BAM, Sorted and Indexed**: We converted the SAM file to BAM format, sorted the BAM file, and created an index using `samtools`.\n","7.  **Variant Calling**: We performed variant calling using `samtools mpileup` and `bcftools call` to identify genetic variations and generated a VCF file.\n","8.  **Previewed VCF File**: We displayed the first few lines of the resulting VCF file to examine its format and content.\n","9.  **Interpreted VCF Header**: We provided a line-by-line interpretation of the header section of the VCF file.\n","10. **Provided Final Conclusion**: We summarized the successful completion of the basic variant calling pipeline and the significance of the VCF file."]}]}